### ISSUE TITLE
Harddisk space on the server

### ISSUE THREAD

**Issue by alleonhardt** [ShadowItaly/clas-digital#261]
At the moment we only have a 100GB of harddisk space left on the server. This means if a lot of uploading is going on we could run out of hard disk space. So we maybe have to work out this problem in the near future.

**Comment by ekoehring** [comment]
Please investigate cost for the needed space and let us know.

**Comment by ekoehring** [comment]
To clarify, we will of course pay for what is needed!

**Comment by georgbuechner** [comment]
Here is the pricing list: https://www.rz.uni-frankfurt.de/46035891/Virtuelle_Server

Additional disc-space is 15€ a month for 1TB and 5€/ month for 300GB.

**Comment by sladen** [comment]
Going for the additional 1TB make sense.  In parallel we should also understand *where* the current disk usage is; there is ~950 GB used, with ~30GB free:

```sh
$ df -h /
Filesystem                         Size  Used Avail Use% Mounted on
/dev/mapper/ubuntu--vg-ubuntu--lv  1,1T  949G   31G  97% /
```
across five main directories:

```sh
$ du -sch /home/clas-digital/srv/{data,{python-clas/web,notneeded/Menard2/install/root}/books} /etc/clas-digital{,-devel}/web/books
192G	/home/clas-digital/srv/data
159G	/home/clas-digital/srv/python-clas/web/books
157G	/home/clas-digital/srv/notneeded/Menard2/install/root/books
268G	/etc/clas-digital/web/books
159G	/etc/clas-digital-devel/web/books
933G	total
```

with the 400G from the previous year:

```sh
$ ls --time-style=long-iso -ltdr /home/clas-digital/srv/{data,{python-clas/web,notneeded/Menard2/install/root}/books} /etc/clas-digital{,-devel}/web/books | awk '{print $6,$8}'
2019-02-05 /home/clas-digital/srv/data
2019-06-10 /home/clas-digital/srv/notneeded/Menard2/install/root/books
2020-06-14 /home/clas-digital/srv/python-clas/web/books
2020-06-21 /etc/clas-digital-devel/web/books
2020-06-21 /etc/clas-digital/web/books
```

All first pages in the original numbering system (`page00001.jpg`) across all directories is ~1.5G:

```sh
find /home/clas-digital/srv/{data,{python-clas/web,notneeded/Menard2/install/root}/books} /etc/clas-digital{,-devel}/web/books -type f -name page00001.jpg | xargs du -ch | tail -n1
1,4G total
```

```sh
find /home/clas-digital/srv/{data,{python-clas/web,notneeded/Menard2/install/root}/books} /etc/clas-digital{,-devel}/web/books -type f -name page00001.jpg | xargs md5sum | sort | uniq -w32 | awk '{print $2}' | xargs du -ch | tail -n1
300M total
```

Therefore, by using hard links (multiple filenames pointing to the same physical file _contents_) it would be possible to reduce the disk space usage—but also reducing the number of "hot spare" backups.

**Comment by rolandborgards** [comment]
Gerne 1TB erweitern!

**Comment by PaulMueller9x** [comment]
Is the expansion complete now? Because I'm repeatedly getting an error when uploading, which I suppose is from missing space.

**Comment by sladen** [comment]
@ShadowItaly @georgbuechner: (when) was the 1TB ordered?  Is there a reference number that we can chase?

(@PaulMueller9x is blocked from doing uploads).

**Comment by PaulMueller9x** [comment]
Uploading is working again. Thanks!

**Comment by sladen** [comment]
~160GB was made available by pointing the two oldest copies to the same file data on disk:

```sh
for i in `seq -w 1 1292` ; do find /home/clas-digital/srv/data /home/clas-digital/srv/python-clas/web/books/ -name "*$i.jpg" -print0 | xargs -0 sudo -u clas-digital rdfind -makehardlinks true ; done
```

(This means that there is one less independent backup).

**Comment by alleonhardt** [comment]
We send the application for one more TB harddisk space to Mr. Borgards. The additional space should be available soon.

**Comment by alleonhardt** [comment]
Problem solved see #278
